src/pages/trader-tools/common/_utility.tsx

```diff
@@ -103,7 +102,7 @@ export const getPnlMarginCommon = (values, action) => {
             return toFixed(stop_loss_pip_formula)
         }
         case 'getTakeProfitPip': {
-            const take_profit_pip_formula = Math.abs(takeProfitLevel - assetPrice) / pointValue
+            const take_profit_pip_formula = Math.abs(stopLossLevel - assetPrice) / pointValue
```

Comment:
Are you sure this works?

Reply:
I've tested this, and it's working.
I've compared Pnl for margin calculator in test link with deriv.com. It still shows the right result with this change.